src/integration-test/java/com/commercetools/sync/inventories/InventoryIntegrationTestUtils.java

```diff
@@ -0,0 +1,223 @@
+
+        final FieldDefinition fieldDefinition = FieldDefinition
+            .of(StringFieldType.of(), CUSTOM_FIELD_NAME, LocalizedString.of(Locale.ENGLISH, CUSTOM_FIELD_NAME), false);
+        final TypeDraft typeDraft = TypeDraftBuilder.of(CUSTOM_TYPE, LocalizedString.of(Locale.ENGLISH, CUSTOM_TYPE),
+            Collections.singleton(InventoryEntry.resourceTypeId())).fieldDefinitions(singletonList(fieldDefinition))
+            .build();
+        CTP_TARGET_CLIENT.execute(TypeCreateCommand.of(typeDraft)).toCompletableFuture().join();
+    }
+
+    /**
+     * Tries to fetch inventory entry of {@code sku} and {@code supplyChannel} using {@code sphereClient}.
+     *
+     * @param sphereClient sphere client used to execute requests
+     * @param sku sku of requested inventory entry
+     * @param supplyChannel optional reference to supply channel of requested inventory entry
+     * @return {@link Optional} which may contain inventory entry of {@code sku} and {@code supplyChannel}
+     */
+    public static Optional<InventoryEntry> getInventoryEntryBySkuAndSupplyChannel(@Nonnull final SphereClient
+                                                                                      sphereClient,
+                                                                                  @Nonnull final String sku,
+                                                                                  @Nullable final Reference<Channel>
+                                                                                      supplyChannel) {
+        InventoryEntryQuery query = InventoryEntryQuery.of().plusPredicates(inventoryEntryQueryModel ->
+            inventoryEntryQueryModel.sku().is(sku));
+        query = supplyChannel == null
+            ? query.plusPredicates(inventoryEntryQueryModel -> inventoryEntryQueryModel.supplyChannel().isNotPresent())
+            : query.plusPredicates(inventoryEntryQueryModel -> inventoryEntryQueryModel.supplyChannel()
+            .is(supplyChannel));
+        return sphereClient.execute(query).toCompletableFuture().join().head();
+    }
+
+    /**
+     * Tries to fetch channel of key {@code channelKey} using {@code sphereClient}.
+     *
+     * @param sphereClient sphere client used to execute requests
+     * @param channelKey key of requested channel
+     * @return {@link Optional} which may contain channel of key {@code channelKey}
+     */
+    public static Optional<Channel> getChannelByKey(@Nonnull final SphereClient sphereClient,
+                                                    @Nonnull final String channelKey) {
+        final ChannelQuery channelQuery = ChannelQueryBuilder.of().plusPredicates(channelQueryModel ->
+            channelQueryModel.key().is(channelKey)).build();
+        return sphereClient.execute(channelQuery).toCompletableFuture().join().head();
+    }
+
+    private static InventoryEntryQuery inventoryEntryQuerySupplier() {
```

Comment:
Because I use the `fetchAndProcess` method from `SphereClientUtils`, which expects a query supplier not the query object itself. So I just pass references to these methods, I found it more readable than inline lambda.
But you're right I'll rename them using a convention 'getQueryOfSth' e.g. `getQueryOfAllInventoryEntries`